ceph.spec.in

```diff
@@ -168,7 +168,7 @@ BuildRequires:	libaio-devel
 BuildRequires:	libblkid-devel >= 2.17
 BuildRequires:	libcurl-devel
 BuildRequires:	libcap-ng-devel
-BuildRequires:	libudev-devel
+BuildRequires:	pkgconfig(libudev)
```

Comment:
may i learn which package provides this macro? shall we install the package before installing the dependencies using ceph.spec.in? also, if the package is available on all rpm based distros we support, probably we should just switch over to it when defining "BuildRequires" and "Provides"?

Reply:
pkgconfig() is not a macro (macros in rpm always start with %) - it is simply a provides added to RPMs based on the presence of pkgconfig .pc files.

e.g.:

```
$ rpm -q --provides libudev-devel:
libudev-devel = 244-1.1
libudev-devel(x86-64) = 244-1.1
pkgconfig(libudev) = 244
```

And like any other symbol provided, I can use it in BuildRequires

Reply:
@DimStar77 thanks. could you answer my second question? can we apply this change to other packages which provide .pc files?

Reply:
There are definitively more that can be changed. My personal proposal is to try to match as closely as possible what the buildsystem does/uses: if configure uses pkg_check, you already have the name of what to put in pkgcofig(...)

Reply:
i see. but our cmake script normally only uses pkgconfig as a hint, so even if the corresponding .pc file is missing cmake might be still able to find the .h and .a files.

Reply:
also, FWIW, Findudev.cmake does not use pkgconfig at this moment. see https://github.com/ceph/ceph/blob/master/cmake/modules/Findudev.cmake

Reply:
interesting - well, yes, then relying on the pkgconfig() symbol should really rather be done 'whever possible' - if only, it allows at least for not having to care for the underlying package names used in the various distributions. That in itself is worth a lot :)

If you want, I can try to work up a follow up PR (after this one) with more symbols converted for review. But I'd prefer to decouple it from this one here (as libudev-devel -> pkgconfig(libudev) gives openSUSE an immediate boost in the distro package rebuild chain - others will huch much less direct impact)

Reply:
> If you want, I can try to work up a follow up PR (after this one) with more symbols converted for review. 

thanks. that'd be great if we can be more consistent in this area.

> But I'd prefer to decouple it from this one here (as libudev-devel -> pkgconfig(libudev) gives openSUSE an immediate boost in the distro package rebuild chain - others will huch much less direct impact)

sure!